File: java/src/main/java/fr/sdecout/handson/persistence/converters/IsbnConverter.java
package fr.sdecout.handson.persistence.converters;

import fr.sdecout.handson.rest.shared.Isbn;
import org.jooq.impl.AbstractConverter;

public final class IsbnConverter extends AbstractConverter<String, Isbn> {

    public IsbnConverter() {
        super(String.class, Isbn.class);
    }

    @Override
    public Isbn from(String databaseObject) {
        return new Isbn(databaseObject);
    }

    @Override
    public String to(Isbn userObject) {
        return userObject.compressedValue();
    }

}

File: kotlin/src/main/kotlin/fr/sdecout/handson/persistence/converters/IsbnConverter.kt
package fr.sdecout.handson.persistence.converters

import fr.sdecout.handson.rest.shared.Isbn
import org.jooq.impl.AbstractConverter

class IsbnConverter : AbstractConverter<String, Isbn>(String::class.java, Isbn::class.java) {
    override fun from(databaseObject: String): Isbn = Isbn(databaseObject)
    override fun to(userObject: Isbn): String = userObject.compressedValue
}
